sketch.js

```diff
@@ -0,0 +1,47 @@
+var colony;
+var nest;
+var supply = [];
+
+function setup() {
+  createCanvas(windowWidth, windowHeight);
+  colony = new Colony();
+  nest = new Nest(100,height/6, 200, radians(45));
+
+  //distribution of food
+  for (var i = 0; i < 15; i++) {
+    var food = new Food(random(width), random(height));
+    if (food.pos.dist(nest.pos) > (nest.r)) {
+      supply.push(food);
+    }
+  }
+  //ants anywhere
+  for (var i = 0; i < 15; i++) {
+    var ant = new Ant(random(width), random(height), nest);
+    colony.addAnt(ant);
+  }
+  //ants inside nest
+  var j = 0;
+  while (j < 7) {
+    push();
+    translate(nest.pos.x, nest.pos.y);
+    var ant = new Ant(random(-nest.r, nest.r), random(-nest.r, nest.r), nest);
```

Comment:
This logic seems like it could be encapsulated within the nest itself since it knows what the `r` value is. Then just have a `nest.randomLocation` function or something which will do this for you.